test/extended/util/cli.go

```diff
@@ -136,26 +139,50 @@ func (c *CLI) AsAdmin() *CLI {
 
 // ChangeUser changes the user used by the current CLI session.
 func (c *CLI) ChangeUser(name string) *CLI {
-	adminClientConfig, err := testutil.GetClusterAdminClientConfig(c.adminConfigPath)
-	if err != nil {
-		FatalErr(err)
-	}
-	_, clientConfig, err := testutil.GetClientForUser(adminClientConfig, name)
-	if err != nil {
-		FatalErr(err)
-	}
+	err := kwait.PollImmediate(1*time.Second, 2*time.Minute, func() (bool, error) {
+		adminClientConfig, err := testutil.GetClusterAdminClientConfig(c.adminConfigPath)
+		if err != nil {
+			// if retryable, or if unauthorized, which *could* happen if openshift api
+			// server is momentarily unavailable during install/startup
```

Comment:
i'm wary of us masking this condition.  is there a legitimate reason we consider it "ok" for openshift api to ever be unavailable?

i know killing people's innocent tests when we might have been able to retry and move on feels like collateral damage, but if there's really an issue w/ the api server being flakily available, this might be the best way for us to get a signal on it.

Reply:
There are so many subsequent interactions with the api server in the actual tests, I think we should still catch if the apiserver is unstable if the test actually runs.

In fact, we got such a failure in the e2e-aws-builds run below at https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/22435/pull-ci-openshift-origin-master-e2e-aws-builds/1173  ... which by the way was just a little further down the line in the project setup from what this code change addressed.  So that spoke to the "narrow focus" I mentioned in the description, and that the possible window is in fact wider than suspected.  Literally failed at line 249 instead of 215.

And this time the apiserver statuses were good as opposed to the run noted in this PRs description.

So yeah, as I indicated in the description, I'm not 100% certain we wanted to do this, not knowing how large the unavailability windows might be.  But the notion occurred to me and I thought a PR with such a change was a good discussion incubator.

And this test run revealed another permutation to the problem, at least to my understanding.

I'll close this out.

Reply:
though I'll initiate another run just for data first